AdventOfCode/Utilities.cs

```diff
@@ -17,13 +17,6 @@ public static IEnumerable<int> AllIndexesOf(this string str, string substring)
         }
     }
 
-    public static IEnumerable<(int Index, T Value)> Enumerate<T>(this IEnumerable<T> enumerable)
-    {
-        var i = 0;
-
-        foreach (var e in enumerable)
-        {
-            yield return (i++, e);
-        }
-    }
+    public static IEnumerable<(int Index, T Value)> EnumerateWithIndex<T>(this IEnumerable<T> enumerable) =>
```

Comment:
I refuse to rename this, because `enumerate` is the python method name for doing this, so it makes sense to me...
But the fact that this can be done in Select (though not, apparently, in a foreach directly) is awesome...